### PR TITLE
improve: add metrics write_data_in and sql_data_in

### PR DESCRIPTION
**Pull request by ZuoTiJia**
# Which issue does this PR close?

Related #.

# Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by roseboy-liu** [comment]
This pr is related to @nakahaa ， please check

File: main/src/http/http_service.rs [hunk 1]
```diff
@@ -442,10 +455,15 @@ impl HttpService {
...
+                    let (tenant, db, user, addr) = (
+                        ctx.tenant(),
+                        ctx.database(),
+                        ctx.user_info().desc().name(),
+                        addr.as_str(),
+                    );
 
-                    metrics.writes_inc(tenant, user, db, addr.as_str());
+                    metrics.writes_inc(tenant, user, db, addr);
+                    metrics.write_data_in_inc(tenant, user, db, addr, points_len);
```
Comment (yukkit) [inline main/src/http/http_service.rs:466]:
```suggestion
                    metrics.write_data_in_inc(tenant, user, db, addr, req.len());
```

points_len is the serialized data size of flatbuffer, suggest to use the original data len directly

File: main/src/http/http_service.rs [hunk 2]
```diff
@@ -632,27 +650,39 @@ impl HttpService {
...
+                    let (tenant, user, db, addr) = (
+                        ctx.tenant(),
+                        ctx.database(),
+                        ctx.user_info().desc().name(),
+                        addr.as_str(),
+                    );
 
-                    metrics.writes_inc(tenant, user, db, addr.as_str());
+                    metrics.writes_inc(tenant, user, db, addr);
+                    metrics.write_data_in_inc(tenant, user, db, addr, points_len);
```
Comment (yukkit) [inline main/src/http/http_service.rs:677]:
```suggestion
                    metrics.write_data_in_inc(tenant, user, db, addr, req.len());
```

File: query_server/query/src/extension/physical/transform_rule/table_writer.rs [hunk 3]
```diff
@@ -69,7 +76,13 @@ impl ExtensionPlanner for TableWriterPlanner {
...
                     "TableWriterMergePlanNode has multiple inputs."
                 );
-                let physical_input = physical_inputs[0].clone();
+
+                let batch_size = session_state.config().batch_size();
+                // CoalesceBatchesExec combines small batches into larger batches for more efficient writing
+                let physical_input = Arc::new(CoalesceBatchesExec::new(
+                    physical_inputs[0].clone(),
+                    batch_size,
+                ));
```
Comment (yukkit) [inline query_server/query/src/extension/physical/transform_rule/table_writer.rs:85]: This does not need to be added, the actual write operation is `TableWriterPlanNode `, and `TableWriterMergePlanNode ` just summarizes the write results

**Review by yukkit — APPROVED** [review]

**Review by yukkit — APPROVED** [review]

**Pull request merged**